src/main/java/com/faforever/client/replay/ReplayCardController.java

```diff
@@ -136,13 +149,31 @@ protected void onInitialize() {
                     .map(reviewsSummary -> reviewsSummary.score() / reviewsSummary.numReviews())
             .when(showing));
 
+
+    isReplayWatched.bind(Bindings.createBooleanBinding(() -> {
+      Integer id = entity.map(Replay::id).getValue();
+      ObservableSet<Integer> replayHistory = this.replayHistory.getWatchedReplays();
+      if( id != null && replayHistory != null ){
+        return replayHistory.contains(id);
+      } else {
+        return false;
+      }
+    }, entity, replayHistory.getWatchedReplays()));
+
+
+    isReplayWatched.when(showing).subscribe((oldValue, newValue)-> {
+      if(newValue) {
+        applyWatchedReplayHighlight();
+      }
+    });
```

Comment:
Since we don't use the oldValue we can just use the single value version of subscribe
```suggestion
    isReplayWatched.when(showing).subscribe(value-> {
      if (value) {
        applyWatchedReplayHighlight();
      }
    });
```